appliance.portal configuration crash during proxy_p12s

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->



### Terraform CLI, Terraform appgatesdp Provider Version, and Appgate Appliance version(s)

terraform cli: v1.0.0.0
terraform-provider-appgatesdp:  v6.6.8
appgate appliance version: >= 5.4.0

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->
<!--- Please include your appliance version --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* appgatesdp_appliance

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file.
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

```
│ Error: Plugin did not respond
│
│   with module.appgate_logserver.appgatesdp_appliance.appliance_definition,
│   on modules/appgate_additional_appliance/main.tf line 6, in resource "appgatesdp_appliance" "appliance_definition":
│    6: resource "appgatesdp_appliance" "appliance_definition"{
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-appgatesdp_v0.6.8 plugin:
panic: runtime error: index out of range [0] with length 0



goroutine 52 [running]:
github.com/appgate/terraform-provider-appgatesdp/appgate.flattenAppliancePortalProxyp12s(0xf73820, {0xc0001ec1e0, 0x1, 0x7})
                github.com/appgate/terraform-provider-appgatesdp/appgate/resource_appgate_appliance.go:1726 +0x4cf
github.com/appgate/terraform-provider-appgatesdp/appgate.resourceAppgateApplianceRead(0xc000327080, {0xf83660, 0xc000264230})
                github.com/appgate/terraform-provider-appgatesdp/appgate/resource_appgate_appliance.go:1686 +0x2125
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x10a9bf7, {0x126c588, 0xc000501bc0}, 0x24, {0xf83660, 0xc000264230})
                github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:335 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000519880, {0x126c588, 0xc000501bc0}, 0xc0004fe9a0, {0xf83660, 0xc000264230})
                github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:624 +0x35b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0000a3698, {0x126c588, 0xc000501bc0}, 0xc000501c00)
                github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/grpc_provider.go:575 +0x4bd
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ReadResource(0xc000072ca0, {0x126c630, 0xc00059f170}, 0xc000324b40)
                github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:298 +0x1fb
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x105c7e0, 0xc000072ca0}, {0x126c630, 0xc00059f170}, 0xc000324ae0, 0x0)
                github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000578380, {0x127bc60, 0xc000110900}, 0xc000696100, 0xc00053cd20, 0x19bea50, 0x0)
                google.golang.org/grpc@v1.32.0/server.go:1194 +0xc8f
google.golang.org/grpc.(*Server).handleStream(0xc000578380, {0x127bc60, 0xc000110900}, 0xc000696100, 0x0)
                google.golang.org/grpc@v1.32.0/server.go:1517 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
                google.golang.org/grpc@v1.32.0/server.go:859 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
                google.golang.org/grpc@v1.32.0/server.go:857 +0x294
Error: The terraform-provider-appgatesdp_v0.6.8 plugin crashed!

 

This is always indicative of a bug within the plugin. It would be immensely

helpful if you could report the crash with the plugin's maintainers so that it

can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

<!--- What should have happened? --->

### Actual Behavior

<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

--- a/appgate/resource_appgate_appliance.go
+++ b/appgate/resource_appgate_appliance.go
@@ -1722,7 +1722,7 @@ func flattenAppliancePortalProxyp12s(local map[string]interface{}, p12s []openap
 		raw["verify_upstream"] = p12.GetVerifyUpstream()
 		raw["subject_name"] = p12.GetSubjectName()
 		// content, and password not always known, not included in the response body
-		if state, ok := local["proxy_p12s"].([]interface{}); ok {
+		if state, ok := local["proxy_p12s"].([]interface{}); ok && state[k] != nil {
 			stateRow := state[k].(map[string]interface{})
 			raw["content"] = stateRow["content"].(string)
 			raw["password"] = stateRow["password"].(string)

